src/Fixer/LanguageConstruct/FunctionToConstantFixer.php

```diff
@@ -0,0 +1,189 @@
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author SpacePossum
+ */
+final class FunctionToConstantFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    /**
+     * @var array<string, string>
+     */
+    private static $defaultConfiguration = array(
+        'phpversion' => 'PHP_VERSION',
+        'php_sapi_name' => 'PHP_SAPI',
+        'pi' => 'M_PI',
+    );
+
+    /**
+     * @var array<string, string>
+     */
+    private $configuration;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        if (null === $configuration) {
+            $this->configuration = self::$defaultConfiguration;
+
+            return;
+        }
+
+        $this->configuration = array();
+        foreach ($configuration as $key) {
+            if (!is_string($key) || !array_key_exists($key, self::$defaultConfiguration)) {
+                throw new InvalidFixerConfigurationException($this->getName(), sprintf('Configuration key must any of "%s".', implode('", "', self::$defaultConfiguration)));
+            }
+
+            $this->configuration[strtolower($key)] = self::$defaultConfiguration[$key];
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        for ($index = $tokens->count() - 4; $index > 0; --$index) {
+            $candidate = $this->isReplaceCandidate($tokens, $index);
+            if (false === $candidate) {
+                continue;
+            }
+
+            $this->fixFunctionCallToConstant(
+                $tokens,
+                $index,
+                $candidate[0], // brace open
+                $candidate[1], // brace close
+                $candidate[2]  // replacement
+            );
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Replace core functions calls returning constants with the constants.',
+            array(new CodeSample("<?php\necho phpversion();\necho pi();\necho php_sapi_name();")),
+            '',
+            'Configure any of `phpversion`, `php_sapi_name` and `pi`.',
+            array_keys(self::$defaultConfiguration),
+            'Risky when the any of the configured functions to replace are overridden.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should run before NativeFunctionCasingFixer
+        // must run before NoExtraConsecutiveBlankLinesFixer, NoSinglelineWhitespaceBeforeSemicolonsFixer, NoTrailingWhitespaceFixer and NoWhitespaceInBlankLineFixer
+        // must run after NoSpacesAfterFunctionNameFixer and NoSpacesInsideParenthesisFixer
```

Comment:
we can ignore the white space, but that way we would have diff. end result based on which fixers are run first